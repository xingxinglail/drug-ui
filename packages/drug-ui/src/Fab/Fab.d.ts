import * as React from 'react';
import { PropTypes as ComponentPropTypes } from '..';
import { ButtonBaseProps } from '../ButtonBase';
export interface FabProps extends React.ButtonHTMLAttributes<HTMLButtonElement>, ButtonBaseProps {
    color?: ComponentPropTypes.Color;
    size?: ComponentPropTypes.Size;
    href?: string;
    disabled?: boolean;
    ref?: React.Ref<HTMLButtonElement>;
}
declare const Fab: React.FC<FabProps>;
export default Fab;
