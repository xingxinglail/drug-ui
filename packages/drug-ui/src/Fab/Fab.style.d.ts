import { Theme } from '../styles';
export declare const styles: (theme: Theme) => Record<string, any>;
