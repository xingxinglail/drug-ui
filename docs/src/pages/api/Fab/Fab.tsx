import * as React from 'react';
import MarkdownDocs from '../../../components/MarkdownDocs';
import markdown from './index.md';

export default function Page () {
    return <MarkdownDocs markdown={ markdown } />;
}
