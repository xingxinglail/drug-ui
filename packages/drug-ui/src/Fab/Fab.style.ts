import { Style, Styles } from 'jss';
import { Theme } from '../styles';

const root: Style = (theme: Theme) => ({
    width: 56,
    height: 56,
    padding: 0,
    fontSize: theme.typography.fontSize.large,
    minWidth: 0,
    boxShadow: '0px 3px 5px -1px rgba(0,0,0,0.2), 0px 6px 10px 0px rgba(0,0,0,0.14), 0px 1px 18px 0px rgba(0,0,0,0.12)',
    boxSizing: 'border-box',
    minHeight: 36,
    transition: 'background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms',
    fontWeight: 500,
    lineHeight: 1.75,
    borderRadius: '50%',
    backgroundColor: '#e0e0e0',
    '&:hover': {
        textDecoration: 'none',
        backgroundColor: '#d5d5d5'
    },
    '&$disabled': {
        color: 'rgba(0, 0, 0, 0.26)',
        boxShadow: 'none',
        backgroundColor: 'rgba(0, 0, 0, 0.12)'
    }
});

const primary: Style = (theme: Theme) => ({
    color: theme.palette.primary.contrastText,
    backgroundColor: theme.palette.primary.main,
    '&:hover': {
        backgroundColor: '#115293'
    }
});

const secondary: Style = (theme: Theme) => ({
    color: theme.palette.secondary.contrastText,
    backgroundColor: theme.palette.secondary.main,
    '&:hover': {
        backgroundColor: '#9a0036'
    }
});

const colorInherit: Style = {
    color: 'inherit'
};

const sizeSmall: Style = {
    width: 40,
    height: 40
};

const sizeMedium: Style = {
    width: 48,
    height: 48
};

const disabled: Style = {};

export const styles = (theme: Theme): Styles => ({
    root: root(theme),
    primary: primary(theme),
    secondary: secondary(theme),
    sizeSmall,
    sizeMedium,
    colorInherit,
    disabled
});
