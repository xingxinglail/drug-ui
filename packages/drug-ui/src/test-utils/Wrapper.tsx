import * as React from 'react';
import { ThemeProvider } from '../index';

interface Props {
    children?: any;
}

const Wrapper: React.FC<Props> = (props) => {
    return (
        <ThemeProvider>
            { props.children }
        </ThemeProvider>
    );
};

export default Wrapper;
