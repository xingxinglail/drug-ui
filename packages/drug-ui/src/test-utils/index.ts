export { default as Wrapper } from './Wrapper';
export { getClasses } from './getClasses';

export const wait = (delay = 50) => {
    return new Promise(resolve => setTimeout(resolve, delay));
};
