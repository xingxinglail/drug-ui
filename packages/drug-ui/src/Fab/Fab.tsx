import * as React from 'react';
import * as PropTypes from 'prop-types';
import { PropTypes as ComponentPropTypes } from '..';
import classnames from 'classnames';
import { capitalize } from '../utils';
import { styles } from './Fab.style';
import { createUseStyles } from '../styles';
import ButtonBase, { ButtonBaseProps } from '../ButtonBase';

export interface FabProps extends React.ButtonHTMLAttributes<HTMLButtonElement>, ButtonBaseProps {
    color?: ComponentPropTypes.Color;
    size?: ComponentPropTypes.Size;
    disabled?: boolean;
    ref?: React.Ref<HTMLButtonElement>
}

type FabClassProps = 'root' | 'primary' | 'secondary' | 'colorInherit' | 'small' | 'medium' | 'disabled';

const name = 'Fab';

const useStyles = createUseStyles<FabClassProps>(styles, name);

const Fab: React.FC<FabProps> = React.forwardRef<HTMLButtonElement, FabProps>((props, ref) => {
    const { className, color = 'default', size = 'large', disabled, component, href, children, ...rest } = props;
    const classes = useStyles();

    const classNames = classnames(
        classes.root,
        {
            [classes[color]]: color !== 'inherit' && color !== 'default',
            [classes[`size${ capitalize(size) }`]]: size !== 'large',
            [classes.colorInherit]: color === 'inherit',
            [classes.disabled]: disabled
        }
    );

    return (
        <ButtonBase
            className={ classNames }
            component={ component }
            ref={ ref }
            href={ href }
            disabled={ disabled }
            { ...rest }>
            { children }
        </ButtonBase>
    );
});

Fab.displayName = name;

Fab.defaultProps = {
    color: 'default',
    size: 'large'
};

Fab.propTypes = {
    color: PropTypes.oneOf<ComponentPropTypes.Color>(['default', 'primary', 'secondary', 'inherit']),
    size: PropTypes.oneOf<ComponentPropTypes.Size>(['small', 'medium', 'large'])
};

export default Fab;
