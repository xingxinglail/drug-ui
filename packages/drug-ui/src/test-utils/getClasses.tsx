import * as React from 'react';
import * as enzyme from 'enzyme';
import Wrapper from './Wrapper';

export const getClasses = (element: React.ReactElement) => {
    // @ts-ignore
    const { useStyles } = element.type;
    let classes;

    function Listener () {
        classes = useStyles();
        return null;
    }

    enzyme.mount(<Wrapper><Listener /></Wrapper>);
    return classes;
};
