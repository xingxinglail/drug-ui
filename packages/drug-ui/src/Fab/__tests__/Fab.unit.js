import * as renderer from 'react-test-renderer';
import * as React from 'react';
import * as enzyme from 'enzyme';
import Fab from '..';
import { Wrapper } from '@drug-ui/core/test-utils';

const mount = enzyme.mount;

describe('<Fab />', () => {

    it('render successful', () => {
        const json = renderer.create(<Wrapper><Fab /></Wrapper>).toJSON();
        expect(json).toMatchSnapshot();
    });

    it('可以设置 className', () => {
        const wrapper = mount(<Wrapper><Fab className="Fab Fab2">Hello World</Fab></Wrapper>);
        const classList = wrapper.getDOMNode().classList;
        expect(classList.contains('Fab')).toBeTruthy();
        expect(classList.contains('Fab2')).toBeTruthy();
    });

    it('默认按钮', () => {
        const wrapper = mount(<Wrapper><Fab><span>1</span></Fab></Wrapper>);
        const classList = wrapper.getDOMNode().classList;
        expect(classList.contains('DuiButtonBase-root')).toBeTruthy();
        expect(classList.contains('DuiFab-root')).toBeTruthy();
        expect(classList.length).toBe(2);
    });

    it('设置 color = inherit', () => {
        const wrapper = mount(<Wrapper><Fab color="inherit" /></Wrapper>);
        const classList = wrapper.getDOMNode().classList;
        expect(classList.contains('DuiButtonBase-root')).toBeTruthy();
        expect(classList.contains('DuiFab-root')).toBeTruthy();
        expect(classList.contains('DuiFab-colorInherit')).toBeTruthy();
        expect(classList.length).toBe(3);
    });

    it('设置 color = primary', () => {
        const wrapper = mount(<Wrapper><Fab color="primary" /></Wrapper>);
        const classList = wrapper.getDOMNode().classList;
        expect(classList.contains('DuiButtonBase-root')).toBeTruthy();
        expect(classList.contains('DuiFab-root')).toBeTruthy();
        expect(classList.contains('DuiFab-primary')).toBeTruthy();
        expect(classList.length).toBe(3);
    });

    it('设置 color = secondary', () => {
        const wrapper = mount(<Wrapper><Fab color="secondary" /></Wrapper>);
        const classList = wrapper.getDOMNode().classList;
        expect(classList.contains('DuiButtonBase-root')).toBeTruthy();
        expect(classList.contains('DuiFab-root')).toBeTruthy();
        expect(classList.contains('DuiFab-secondary')).toBeTruthy();
        expect(classList.length).toBe(3);
    });

    it('设置 color = xxx', () => {
        const wrapper = mount(<Wrapper><Fab color="xxx" /></Wrapper>);
        const classList = wrapper.getDOMNode().classList;
        expect(classList.contains('DuiButtonBase-root')).toBeTruthy();
        expect(classList.contains('DuiFab-root')).toBeTruthy();
        expect(classList.contains('DuiFab-colorInherit')).toBeFalsy();
        expect(classList.contains('DuiFab-primary')).toBeFalsy();
        expect(classList.contains('DuiFab-secondary')).toBeFalsy();
        expect(classList.length).toBe(2);
    });

    it('设置 size = small', () => {
        const wrapper = mount(<Wrapper><Fab size="small" /></Wrapper>);
        const classList = wrapper.getDOMNode().classList;
        expect(classList.contains('DuiButtonBase-root')).toBeTruthy();
        expect(classList.contains('DuiFab-root')).toBeTruthy();
        expect(classList.contains('DuiFab-sizeSmall')).toBeTruthy();
        expect(classList.length).toBe(3);
    });

    it('设置 size = medium', () => {
        const wrapper = mount(<Wrapper><Fab size="medium" /></Wrapper>);
        const classList = wrapper.getDOMNode().classList;
        expect(classList.contains('DuiButtonBase-root')).toBeTruthy();
        expect(classList.contains('DuiFab-root')).toBeTruthy();
        expect(classList.contains('DuiFab-sizeMedium')).toBeTruthy();
        expect(classList.length).toBe(3);
    });

    it('设置 size = xxx', () => {
        const wrapper = mount(<Wrapper><Fab size="xxx" /></Wrapper>);
        const classList = wrapper.getDOMNode().classList;
        expect(classList.contains('DuiButtonBase-root')).toBeTruthy();
        expect(classList.contains('DuiFab-root')).toBeTruthy();
        expect(classList.contains('DuiFab-sizeSmall')).toBeFalsy();
        expect(classList.contains('DuiFab-sizeMedium')).toBeFalsy();
        expect(classList.contains('DuiFab-sizeLarge')).toBeFalsy();
        expect(classList.length).toBe(2);
    });
});
