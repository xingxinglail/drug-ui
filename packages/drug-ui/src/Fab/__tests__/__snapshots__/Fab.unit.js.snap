// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Fab /> render successful 1`] = `
<button
  className="DuiButtonBase-root DuiFab-root"
  disabled={false}
>
  <span
    className="DuiRipple-root"
    onDragLeave={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
  />
</button>
`;
